doc-requirements.in

```diff
@@ -1,16 +1,63 @@
 codespell
-git+https://github.com/flyteorg/furo@main
-sphinx
+furo
+flytekit
+flytekitplugins-deck-standard
+flytekitplugins-kfpytorch
+flytekitplugins-sqlalchemy
+gitpython
+ipython!=8.7.0
+jupytext
+myst-nb
+plotly
+psycopg2-binary
+recommonmark
+sphinx==4.5.0
 sphinx-prompt
+sphinx-click
 sphinx-code-include
 sphinx-autoapi
 sphinx-copybutton
 sphinxext-remoteliteralinclude
 sphinx-issues
 sphinx_fontawesome
 sphinx-panels
+sphinx_markdown_tables
 sphinxcontrib-mermaid
 sphinxcontrib-video
 sphinxcontrib-youtube==1.2.0
 sphinx-tabs
-sphinx-tags
+sphinx-tags==0.2.1
+
+# Packages for Plugin docs
```

Comment:
Most of these dependencies are from `flytesnacks`, which makes the docs a little bit harder to build.

For example, I usually need to remove `vaex` because it's build system fails on a M1 mac.

Reply:
many of them are also from flytekit to render the [plugin API reference](https://docs.flyte.org/projects/flytekit/en/latest/plugins/index.html).

Reply:
>  I usually need to remove vaex because it's build system fails on a M1 mac.

What error are you getting?

Reply:
By looking at https://pypi.org/project/vaex-core/#files, `vaex-core` does not have wheels built for Python 3.11. Thus, it triggers local compilation on my Python 3.11 environment.

I can work around it by using `conda-forge` for `vaex` and other dependencies. If I start working with the repo more, I'll likely define a custom conda `environment.yml` file to develop on.

BTW https://github.com/conda/conda-lock is a nice lockfile tool if you use conda + pip.